# include/vlp16.launch.py
cabot:
  speed_visualize_node:
    ros__parameters:
      cmd_vel_topic: /cabot/cmd_vel
      visualize_topic: /cabot/poi

  clearing_tf_node:
    ros__parameters:
      target_rate: 20
      target_frame: velodyne_base_link

  odom_adapter_node:
    ros__parameters:
      odom_input: /cabot/odometry/filtered
      odom_output: /odom
      odom_frame: odom
      base_frame: base_footprint
      offset_frame: base_control_shift
      publish_tf: true
      max_speed: 1.0
      cmd_vel_input: /cmd_vel
      cmd_vel_output: /cabot/cmd_vel_adapter
      target_rate: 20

  odom_adapter_node2:
    ros__parameters:
      odom_input: /cabot/odometry/filtered
      # do nothing with odom topic
      odom_output: /odom_unused
      # copy odom tf
      odom_frame: local/odom
      base_frame: local/base_footprint
      offset_frame: local/base_footprint
      publish_tf: true
      max_speed: 1.0
      # do nothing with cmd_vel
      cmd_vel_input: /cmd_vel_
      cmd_vel_output: /cabot/cmd_vel_adapter_
      target_rate: 20

  lidar_speed_control_node: # old name cabot_e_lidar_speed
    ros__parameters:
      laser_topic: /scan
      visualize_topic: visualize
      limit_topic: /cabot/lidar_speed
      # front obstacle detection
      check_front_obstacle: true
      min_speed: 0.0
      min_distance: 1.0
      limit_factor: 1.0
      front_angle_in_degree: 28.95  # 1.0 * sin(x/2) * 2 = 0.8

  people_speed_control_node: # old name cabot_e_people_speed:
    ros__parameters:
      people_topic: /people
      visualize_topic: /visualize
      limit_topic: /cabot/people_speed
      social_distance_limit_topic: /cabot/social_distance_speed
      pure_velocity_obstacle_limit_topic: /cabot/pure_velocity_obstacle_speed
      combined_speed_limit_topic: /cabot/combined_speed
      odom_topic: /odom
      plan_topic: /plan
      # Social Distancing setting
      # X (front): may need to bigger in larger place
      # Y (side) : need to be small to avoid strange speed down when people are walking through very close
      # Consider an oval radius (x, y)
      social_distance_x: 2.0
      social_distance_y: 0.5
      use_velocity_obstacle: true

  tf_speed_control_node: # old name cabot_e_tf_speed:
    ros__parameters:
      limit_topic: /cabot/tf_speed
      tf_decel_timeout: 0.3
      tf_timeout: 0.5
      min_speed: 0.001

  touch_speed_control_node:
    ros__parameters:
      touch_speed_max_inactive: 0.5

  speed_control_node_touch_true:
    ros__parameters:
      cmd_vel_input: /cabot/cmd_vel_adapter
      cmd_vel_output: /cabot/cmd_vel
      speed_input: ['/cabot/user_speed', '/cabot/lidar_speed', '/cabot/people_speed', '/cabot/tf_speed', '/cabot/queue_speed', '/cabot/map_speed', '/cabot/touch_speed_switched', '/cabot/wheelie_speed']
      speed_limit: [2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0]
      speed_timeout: [-1.0, -1.0, -1.0, -1.0, -1.0, -1.0, 0.5, 0.5]
      complete_stop: [false,false,false,true,true,false,true,false]
      configurable: [true,false,false,false,false,false,false,false]

  speed_control_node_touch_false:  #TODO better way
    ros__parameters:
      cmd_vel_input: /cabot/cmd_vel_adapter
      cmd_vel_output: /cabot/cmd_vel
      speed_input: ['/cabot/user_speed', '/cabot/lidar_speed', '/cabot/people_speed', '/cabot/tf_speed', '/cabot/queue_speed', '/cabot/map_speed', '/cabot/wheelie_speed']
      speed_limit: [2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0]
      speed_timeout: [-1.0, -1.0, -1.0, -1.0, -1.0, -1.0, 0.5, 0.5]
      complete_stop: [false,false,false,true,true,false,true,false]
      configurable: [true,false,false,false,false,false,false,false]

  wheelie_control_node:
    ros__parameters:
      pitch_threshold: -0.15
      imu_topic: /cabot/imu/data
      cmd_vel_topic: /cabot/cmd_vel
      wheelie_speed_topic: /cabot/wheelie_speed
      min_speed: 0.0
      max_speed: 1.0

  ekf_node:
    ros__parameters:
      frequency: 100.0
      sensor_timeout: 0.1
      two_d_mode: true
      odom_frame: odom
      base_link_frame: base_footprint
      world_frame: odom
      initial_state: [
        0.0, 0.0, 0.0,
        0.0, 0.0, 0.0,
        0.0, 0.0, 0.0,
        0.0, 0.0, 0.0,
        0.0, 0.0, 0.0]

      # wheel odometry
      odom0: /cabot/odom_raw
      odom0_config: [
        true , true , false,
        false, false, false,
        false, false, false,
        false, false, false,
        false, false, false]
      odom0_queue_size: 100
      odom0_differential: true

      # imu
      imu0: /cabot/imu/data
      imu0_config: [
        false, false, false,
        true , true , true ,
        false, false, false,
        true , true , true ,
        false, false, false]
      imu0_queue_size: 100
      imu0_differential: false
      imu0_relative: false
      imu0_remove_gravitational_acceleration: true
      print_diagnostics: true
      publish_tf: false
